source/common/event/dispatcher_impl.cc

```diff
@@ -121,15 +122,29 @@ DispatcherImpl::createClientConnection(Network::Address::InstanceConstSharedPtr
                                                          std::move(transport_socket), options);
 }
 
+absl::optional<bool> DispatcherImpl::use_apple_api_for_dns_lookups = absl::nullopt;
```

Comment:
This should use CONSTRUCT_ON_FIRST_USE, or just a local static allocation in the function in question. It also doesn't have to be an optional. It can be a bool which is set on first construction.

Reply:
Sure, happy to simplify the overkill use here, although I like the expressiveness of it :)